### ISSUE TITLE
Document `EditInfo` in `Warning`

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#8342]
### Describe the documentation issue

See: 
- https://github.com/stylelint/stylelint/issues/7192
- https://github.com/stylelint/stylelint/issues/8341

Currently `EditInfo` is not documented.

### What solution would you like to see?

See ESLint docs for inspiration: https://eslint.org/docs/latest/integrate/nodejs-api#-editinfo-type

I like that ESLint also documents the operation that can be run to apply the `EditInfo` data: 

```js
 sourceCodeText.slice(0, edit.range[0]) + edit.text + sourceCodeText.slice(edit.range[1])
```

`EditInfo` will only be recorded when `computeEditInfo` was set.

Only a single `EditInfo` will be recorder for a specific region in source code. If multiple report ranges overlap only the first will contain `EditInfo`.
